tests/phpunit/tests/block-template-utils.php

```diff
@@ -471,4 +473,79 @@ public function test_inject_ignored_hooked_blocks_metadata_attributes_into_templ
 			'The hooked block was not injected into the anchor block\'s ignoredHookedBlocks metadata.'
 		);
 	}
+
+	/**
+	 * @ticket 60754
+	 *
+	 * @covers inject_ignored_hooked_blocks_metadata_attributes
+	 */
+	public function test_inject_ignored_hooked_blocks_metadata_attributes_applies_filter_correctly() {
+		global $wp_current_filter;
+		// Mock currently set filter.
+		// The $wp_current_filter global is reset during teardown by the unit test base class.
+		$wp_current_filter[] = 'rest_pre_insert_wp_template';
+
+		$action = new MockAction();
+		add_filter( 'hooked_block_types', array( $action, 'filter' ), 10, 4 );
+
+		$id      = self::TEST_THEME . '//' . 'my_template';
+		$request = new WP_REST_Request( 'POST', '/wp/v2/templates/' . $id );
+		$request->set_param( 'id', $id );
+
+		$changes                = new stdClass();
+		$changes->ID            = self::$template_post->ID;
+		$changes->post_type     = 'wp_template';
+		$changes->post_author   = 123;
+		$changes->post_name     = 'my-updated-template';
+		$changes->post_title    = 'My updated Template';
+		$changes->post_content  = '<!-- wp:tests/anchor-block -->Hello<!-- /wp:tests/anchor-block -->';
+		$changes->post_excerpt  = 'Displays a single post on your website unless a custom template...';
+		$changes->post_status   = 'publish';
+		$changes->post_modified = '2021-07-01 12:00:00'; // FIXME
+		$changes->meta_input    = array(
+			'origin'           => 'custom',
+			'is_wp_suggestion' => true,
+		);
+		$changes->tax_input     = array(
+			'wp_theme' => self::TEST_THEME,
+		);
```

Comment:
Let's reduce this to a minimum number of fields that corresponds to what the controller's `prepare_item_for_database` [would set](https://github.com/WordPress/wordpress-develop/blob/62c807749ec86ee4f213df59f39e7c79b564e3b8/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L556-L569) for this case (where a DB object already exists).